pkg/odo/cli/debug/portforward.go

```diff
@@ -0,0 +1,146 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+
+	componentlabels "github.com/openshift/odo/pkg/component/labels"
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/debug"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	k8sgenclioptions "k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+)
+
+// PortForwardOptions contains all the options for running the port-forward cli command.
+type PortForwardOptions struct {
+	Namespace string
+	Address   []string
+	PortPair  string
+
+	localPort  int
+	contextDir string
+
+	PortForwarder *debug.DefaultPortForwarder
+	StopChannel   chan struct{}
+	ReadyChannel  chan struct{}
+	*genericclioptions.Context
+	localConfigInfo *config.LocalConfigInfo
+}
+
+var (
+	portforwardLong = templates.LongDesc(`
+			Forward a local port to a remote port on the pod where the application is listening for a debugger.
+
+			By default the local port and the remote port will be same but that can be changed using --local-port.  		  
```

Comment:
We're telling how local port can be changed but not telling how remote port can be changed. Or it cannot be changed? Or it can be changed but we don't support that now? Or it doesn't need to be mentioned here (if yes, why)?

Reply:
It needs to be mentioned here. Good point.

Reply:
resolved